scripts/biomcli

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2011-2015, The BIOM Format Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+import click
+import biom.cli
+
+@click.group()
+@click.version_option(version=biom.__version__)
+def cli():
+    pass
+
+@cli.command(name='summarize-table')
+@click.option( '-i', '--input-fp', required=True,
+              type=click.Path(exists=True, dir_okay=False),
+              help='The input BIOM table')
+@click.option('-o', '--output-fp', default=None, type=click.Path(),
+              help='An output file-path')
+@click.option('--qualitative', default=False, is_flag=True,
+              help="Present counts as number of unique observation ids per"
+                    "sample, rather than counts of observations per sample.")
+@click.option('--observations', default=False, is_flag=True,
+              help="Summarize over observations")
+def summarize_table(input_fp, output_fp, qualitative, observations):
+    """Summarize sample or observation data in a BIOM table
+
+    Provides details on the observation counts per sample, including summary
+    statistics, as well as metadata categories associated with samples and
+    observations.
+
+    Example usage:
+
+    Write a summary of table.biom to table_summary.txt:
+
+    $ biom summarize-table -i table.biom -o table_summary.txt
+
+    """
+    result = biom.cli.summarize_table(input_fp, qualitative, observations)
+    if output_fp:
+        with open(output_fp, 'w') as fh:
+            fh.write(result)
+    else:
+        click.echo(result)
```

Comment:
I don't _think_ there is a test for this, but at one point we added in support so that `biom summarize-table` could be piped to other commands like `head`. It would be great if this were still supported. It probably is given click's aggressive POSIX compliance, but just making a note about it.